support/util/expose.go

```diff
@@ -42,3 +42,23 @@ func UseDedicatedDNSForKASByHC(hc *hyperv1.HostedCluster) bool {
 		// and later is used to annotate the route so the external DNS controller can watch it.
 		apiServerService.Route != nil && apiServerService.Route.Hostname != ""
 }
+
+func ServiceExternalDNSHostname(hcp *hyperv1.HostedControlPlane, serviceType hyperv1.ServiceType) string {
+	// external DNS hostname can only be set when HCP is Public
+	if !IsPublicHCP(hcp) {
+		return ""
+	}
```

Comment:
We will also not be able to reach the KAS is the user sets `APIAllowedCIDRBlocks`.  We need to return an empty string in that case as well.

Reply:
added APIAllowedCIDRBlocks check, is that what you meant?